lib/postgrex/extensions/pg_extension.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Postgrex.Extensions.PGExtension do
+  @moduledoc false
+  alias Postgrex.TypeInfo
+  import Postgrex.BinaryUtils
+  use Postgrex.BinaryExtension, type: "citext", type: "hstore"
+
+  def encode(%TypeInfo{type: "citext"}, bin, _, _) when is_binary(bin),
+    do: bin
+  def encode(%TypeInfo{type: "hstore"}, map, _, _),
+    do: encode_hstore(map)
+
+  def decode(%TypeInfo{type: "citext"}, bin, _, _),
```

Comment:
I forget why `citext` is here.. it should be moved to raw because it doesnt need any encode/decde and then we dont use type info. This extension would then be renamed hstore.